### ISSUE TITLE
Kotlinx-IO support

### ISSUE THREAD

**Issue by GazimSoliev** [vinceglb/FileKit#114]
Hello, will this library support kotlinx-io or okio?

**Comment by santiwanti** [comment]
I will look into this. First I need to figure out which to support or if it's worth the trouble of supporting both.

**Comment by GazimSoliev** [comment]
> I will look into this. First I need to figure out which to support or if it's worth the trouble of supporting both.

Of course, kotlinx-io, it's official library by Kotlin, this lib based on okio

**Comment by GazimSoliev** [comment]
in Kotlinx-IO: Path is File in JVM, Source is InputStream in JVM and Sink is OutputStream in JVM, to read, to write and to get data use FileSystem

**Comment by vinceglb** [comment]
Hi @GazimSoliev!

We are working on a big update of FileKit that integrates kotlinx-io in [PlatformFile](https://github.com/vinceglb/FileKit/blob/dc21f609dee5498fc2858c272a946407ce5c5fcb/filekit-core/src/nonWebMain/kotlin/io/github/vinceglb/filekit/PlatformFile.nonWeb.kt#L35) and some new features. It's still in development, but it's usable now! If you are interested in trying out the new features, you can follow the steps indicated in this comment:

https://github.com/vinceglb/FileKit/issues/143#issuecomment-2425180833

If you have any questions or feedback, I'm really interested to hear from you!

**Comment by vinceglb** [comment]
Now, FileKit version `0.10.0-beta01` has a built-in kotlinx-io integration. It enables you to read and write files like you want with kotlinx.io.Path, Sink and Source.

Read more here: https://filekit.mintlify.app/core/platform-file#converting-to-kotlinx-io
